app-backend/common/src/main/scala/cache/CacheClient.scala

```diff
@@ -0,0 +1,83 @@
+package com.azavea.rf.common.cache
+
+import java.util.concurrent.Executors
+
+import net.spy.memcached._
+
+import scala.concurrent._
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import com.azavea.rf.common.Config
+import cats.data._
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import com.typesafe.scalalogging.LazyLogging
+
+object CacheClientThreadPool {
+  implicit lazy val ec: ExecutionContext =
+    ExecutionContext.fromExecutor(
+      Executors.newFixedThreadPool(
+        Config.memcached.threads,
+        new ThreadFactoryBuilder().setNameFormat("cache-client-%d").build()
+      )
+    )
+}
+
+class CacheClient(client: => MemcachedClient) extends LazyLogging {
+
+  import CacheClientThreadPool._
+
+  val cacheEnabled = Config.memcached.enabled
+
+  def delete(key: String) =
+    if(cacheEnabled) {
+      client.delete(key)
+    }
+
+  def setValue[T](key: String, value: T, ttlSeconds: Int = 0): Unit = {
+    Future {
+      client.set(key, ttlSeconds, value)
+    }
```

Comment:
My initial comment here was that it would be nice to return the `Future` so it could be used in `onComplete`. But looking into it further it looks `set` method returns a Java future anyway: https://github.com/dustin/java-memcached-client/blob/master/src/main/java/net/spy/memcached/MemcachedClient.java#L928

The underlying memachedClient has its own `executorService` which is set from [`ConnectionFactory`](https://github.com/dustin/java-memcached-client/blob/master/src/main/java/net/spy/memcached/MemcachedClient.java#L214) (`KyroConnectionFactory` in this case).

The default value there is for fixed to [Runtime.getRuntime().availableProcessors](https://github.com/dustin/java-memcached-client/blob/master/src/main/java/net/spy/memcached/DefaultConnectionFactory.java#L303)

At a minimum the `Future` use here is not needed.